### PR TITLE
Do not trace authn http request bodies

### PR DESCRIPTION
**Pull request by savonarola**
Fixes [EMQX-11130](https://emqx.atlassian.net/browse/EMQX-11130)

### <samp>🤖 Generated by Copilot at c237ddf</samp>

This pull request enhances the security, logging, and tracing of HTTP authentication and bridge requests, adds a new GitHub workflow for Scorecard analysis, fixes some bugs and flaky tests, and updates the version number of the emqx_bridge_http application. It modifies the files `emqx_authn_http.erl`, `emqx_bridge_http_connector.erl`, `emqx_bridge_http_SUITE.erl`, `scorecard.yaml`, `emqx_bridge_http_connector.hrl`, `emqx_bridge_http.app.src`, and `emqx_bridge_kafka_impl_consumer_SUITE.erl`.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [na] Added tests for the changes
- [na] Added property-based tests for code which performs user input validation
- [x] Changed lines covered in coverage report
- [] Change log has been added to `changes/(ce|ee)/(feat|perf|fix)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [na] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [x] Schema changes are backward compatible

### CODE REVIEW

File: apps/emqx_bridge_http/src/emqx_bridge_http_connector.erl [hunk 1]
```diff
@@ -329,7 +330,7 @@ on_query(
         {error, #{status_code := StatusCode}} ->
             ?SLOG(error, #{
                 msg => "http_connector_do_request_received_error_response",
-                note => "the body will be redacted due to security reasons",
+                note => "the request body will be redacted due to security reasons",
                 request => redact_request(NRequest),
```
Comment (savonarola) [inline apps/emqx_bridge_http/src/emqx_bridge_http_connector.erl:334]: Seems that we already decided not to log request bodies, so I just chose to do that more consistently.

Also the redacted authn request is already logged in the authn provider itself.

**Comment by savonarola** [comment]
Also, a user may choose to send the password in the URL part. However, this is completely insecure in many levels, so I don't think we should try to redact the URL all.

Maybe we should just forbid password interpolation in the URL part instead.

**Comment by zmstone** [comment]
we probably need to trace the parameters used to build the request body (before encoded to JSON).

**Comment by savonarola** [comment]
@zmstone we trace full and secured request inside the provider https://github.com/emqx/emqx/blob/master/apps/emqx_auth_http/src/emqx_authn_http.erl#L73

File: apps/emqx_bridge_http/src/emqx_bridge_http_connector.erl [hunk 2]
```diff
@@ -379,7 +381,8 @@ on_query_async(
         "QUERY_ASYNC",
         "http_connector_received",
         #{
-            request => redact(Request),
+            request => redact_request(Request),
+            note => "the request body will be redacted due to security reasons",
```
[the remark this answers is not in the document]
Comment (zmstone) [reply apps/emqx_bridge_http/src/emqx_bridge_http_connector.erl:385]: maybe use a macro for the `note`

**Review by thalesmg — APPROVED** [review]

**Pull request merged**